sparkmagic/sparkmagic/livyclientlib/endpoint.py

```diff
@@ -21,13 +22,37 @@ def __init__(self, url, auth, username="", password="", implicitly_added=False):
     def __eq__(self, other):
         if type(other) is not Endpoint:
             return False
-        return self.url == other.url and self.username == other.username and self.password == other.password and self.auth == other.auth
+        return self.url == other.url and self.username == other.username and self.password == other.password and self.auth == other.auth and self.ssl_info == other.ssl_info
 
     def __hash__(self):
-        return hash((self.url, self.username, self.password, self.auth))
+        return hash((self.url, self.username, self.password, self.auth, self.ssl_info))
 
     def __ne__(self, other):
         return not self == other
 
     def __str__(self):
         return u"Endpoint({})".format(self.url)
+
+class SSLInfo(object):
```

Comment:
Add a docstring explain what this class does, and what the parameters are.

Reply:
Also, this seems like a good place to start using `attrs` (http://www.attrs.org/en/stable/), it'll make this class much much shorter.